operator/operator.go

```diff
@@ -76,7 +66,10 @@ func ProcessDelete(obj interface{}, operator Operator) error {
 		return microerror.MaskAny(err)
 	}
 
-	desiredState := operator.GetEmptyState()
+	desiredState, err := operator.GetDesiredState(obj)
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
```

Comment:
Since the deletion part now only happens on delete events there is no real need for `GetEmptyState` anymore. In the implementation of the ingress-operator I found it more useful now to provide the real desired state to `GetDeleteState` to sort out what has to be deleted. So here the desired state indicates what should be desired to be deleted. Mapping this to the create event, the desired state indicates what should be created. I like this consistency. The model feels more right now, which also holds true because now it actually works.